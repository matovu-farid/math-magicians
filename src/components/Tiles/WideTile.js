import { Component } from 'react';
import PropTypes from 'prop-types';

export default class WideTile extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  onClick=({ target: { textContent } }) => {
    const name = textContent.trim();
    const { handleButtonClick } = this.props;
    handleButtonClick(name);
  }

  render() {
    const { value } = this.props;
    return (
      <button
        type="button"
        className="tile  wide"
        onClick={this.onClick}
      >
        {value}
      </button>
    );
  }
}

WideTile.propTypes = {
  value: PropTypes.string.isRequired,
  handleButtonClick: PropTypes.func.isRequired,
};
