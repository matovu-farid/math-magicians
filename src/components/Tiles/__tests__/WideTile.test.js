import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import WideTile from '../WideTile';

it('renders correctly', () => {
  const tree = renderer.create(<WideTile value={0} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should display correct value', () => {
  render(<WideTile value={0} />);
  const button = screen.queryByRole('button');

  expect(button).toHaveTextContent('0');
});
