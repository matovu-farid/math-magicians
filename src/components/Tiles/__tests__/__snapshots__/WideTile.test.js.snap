// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="tile  wide"
  onClick={[Function]}
  type="button"
>
  0
</button>
`;
